### ISSUE TITLE
SWA attempt

### ISSUE THREAD

**Issue by sethtroisi** [tensorflow/minigo#280]
I'm trying out SWA

Write-up: https://github.com/glinscott/leela-chess/issues/159

Original post: https://github.com/gcp/leela-zero/issues/814

Code: https://github.com/sethtroisi/minigo/tree/swa?files=1

https://github.com/sethtroisi/minigo/blob/swa/swa.py

I have ringmaster test running right now between swa-2, swa-4, swa-8, swa-10, and 000303-olympus

**Comment by sethtroisi** [comment]
```
minigo-cc-swa-2 v minigo-cc-swa-4 (63/100 games)
board size: 19   komi: 7.5
                  wins              black         white       avg cpu
minigo-cc-swa-2     27 42.86%       11 34.38%     16 51.61%    109.58
minigo-cc-swa-4     36 57.14%       15 48.39%     21 65.62%    109.94
                                    26 41.27%     37 58.73%

minigo-cc-swa-4 v minigo-cc-swa-8 (63/100 games)
board size: 19   komi: 7.5
                  wins              black         white       avg cpu
minigo-cc-swa-4     29 46.03%       15 46.88%     14 45.16%    122.72
minigo-cc-swa-8     34 53.97%       17 54.84%     17 53.12%    122.71
                                    32 50.79%     31 49.21%

minigo-cc-swa-8 v minigo-cc-swa-10 (62/100 games)
board size: 19   komi: 7.5
                   wins              black         white       avg cpu
minigo-cc-swa-8      32 51.61%       18 58.06%     14 45.16%    116.61
minigo-cc-swa-10     30 48.39%       17 54.84%     13 41.94%    116.68
                                     35 56.45%     27 43.55%

minigo-cc-swa-2 v minigo-cc-303 (62/100 games)
board size: 19   komi: 7.5
                  wins              black         white       avg cpu
minigo-cc-swa-2     28 45.16%       18 58.06%     10 32.26%    113.40
minigo-cc-303       34 54.84%       21 67.74%     13 41.94%    113.29
                                    39 62.90%     23 37.10%

minigo-cc-swa-10 v minigo-cc-303 (62/100 games)
board size: 19   komi: 7.5
                   wins              black         white       avg cpu
minigo-cc-swa-10     39 62.90%       15 48.39%     24 77.42%    114.57
minigo-cc-303        23 37.10%       7  22.58%     16 51.61%    114.14
                                     22 35.48%     40 64.52%
```

**Comment by sethtroisi** [comment]
Per LZs implementation it seems like we should run another couple of batches of training so that batch norms can be recalculated

https://github.com/gcp/leela-zero/commit/94f48bb10b20d2267fcd3b9b0531f00c51907909#diff-533b2cfc6eb41b6d5b8954e445851082R453
